### ISSUE TITLE
Backport "Make RabbitMQ a regular Erlang/OTP application" to RabbitMQ 3.8.x

### ISSUE THREAD

**Issue by dumbbell** [rabbitmq/rabbitmq-server#2280]
This issue allows us to track the progress of the backport of rabbitmq/rabbitmq-server#2180 to RabbitMQ 3.8.x.

**Comment by gerhard** [comment]
Looking forward to this getting into 3.8 🎉

**Comment by dumbbell** [comment]
The backport is done and tested successfully locally. I verified the diff between those backport branches and `master` branches to make sure I didn't forget a commit.

The only difference around those changes is related to the product info. Some commits were already backported to `v3.8.x`, but not everything. I'm not adding the missing bits now because it is out-of-scope.

I just fixed the pull request to the broker (#2282): it was based on `master`, not `v3.8.x`. The patch doesn change and I restarted the test on Travis.

**Comment by lukebakken** [comment]
> Some commits were already backported to v3.8.x, but not everything. I'm not adding the missing bits now because it is out-of-scope.

How can we keep track of what is missing?

**Comment by dumbbell** [comment]
> How can we keep track of what is missing?

They are still to be QA'd in `master` and were backported a prematurely. So once everything is finished in `master`, I will take care of backporting everything at once.

**Comment by lukebakken** [comment]
Thanks @dumbbell !

**Comment by michaelklishin** [comment]
Are there more backports pending?

**Comment by dumbbell** [comment]
Yes, at least rabbitmq/rabbitmq-server#2277 which is not ready yet.

**Comment by dumbbell** [comment]
#2277 was merged to `master` and backported to 3.8.x.
